internal/pkg/task/config_runner.go

```diff
@@ -104,3 +107,65 @@ func (r *ConfigRunner) validateDependencies() error {
 
 	return nil
 }
+
+// CheckNonZeroExitCode returns the status of the containers part of the given tasks.
+func (r *ConfigRunner) CheckNonZeroExitCode(tasks []*Task) error {
+	var essentialContainers []string
+	taskDefName := fmt.Sprintf("copilot-%s", r.GroupName)
+	taskDefinition, err := r.Starter.TaskDefinition(taskDefName)
+	if err != nil {
+		return fmt.Errorf("get task definition %s of a task", taskDefName)
+	}
+
+	for _, container := range taskDefinition.ContainerDefinitions {
+		if aws.BoolValue(container.Essential) {
+			essentialContainers = append(essentialContainers, aws.StringValue(container.Name))
+		}
+	}
+
+	taskARNs := make([]string, len(tasks))
+	for idx, task := range tasks {
+		taskARNs[idx] = task.TaskARN
+	}
+	describedTasks, describeErr := r.Starter.DescribeTasks(r.Cluster, taskARNs)
+	if describeErr != nil {
+		return describeErr
+	}
+	// Today we only use one container inside a stand-alone task, shall we keep the looping logic
+	// active for multiple containers or only add this logic when we have multiple containers inside a stand-alone task in the future.
```

Comment:
We might not need these comments 🤔

Reply:
This comment is for the team, I will remove this comment before merging this PR